Core/DatFileParsing/SawyerStreamReader.cs

```diff
@@ -154,50 +154,53 @@ public static (DatFileInfo DatFileInfo, ILocoObject? LocoObject) LoadFullObjectF
 			return new(new DatFileInfo(s5Header, objectHeader), newObj);
 		}
 
+		public static string CStringToString(ReadOnlySpan<byte> data, Encoding enc)
+		{
+			var ptr = 0;
+			while (data[ptr++] != '\0') ;
+			return enc.GetString(data[0..(ptr - 1)]); // do -1 to exclude the \0
+		}
+
+		static Dictionary<LanguageId, string> GetNewLanguageDictionary()
+		{
+			var languageDict = new Dictionary<LanguageId, string>();
+			foreach (var language in Enum.GetValues<LanguageId>())
+			{
+				languageDict.Add(language, string.Empty);
+			}
+			return languageDict;
+		}
+
 		public static (StringTable table, int bytesRead) LoadStringTable(ReadOnlySpan<byte> data, string[] stringNames, ILogger? logger = null)
 		{
 			var stringTable = new StringTable();
 
 			if (data.Length == 0 || stringNames.Length == 0)
 			{
+				logger?.Warning($"No data for language table");
 				return (stringTable, 0);
 			}
 
 			var ptr = 0;
 
 			foreach (var locoString in stringNames)
 			{
-				stringTable.Table.Add(locoString, []);
+				// init language table
+				stringTable.Table.Add(locoString, GetNewLanguageDictionary());
 				var languageDict = stringTable[locoString];
 
-				// add empty strings for every single language
-				foreach (var language in Enum.GetValues<LanguageId>())
-				{
-					languageDict.Add(language, string.Empty);
-				}
-
-				for (; ptr < data.Length && data[ptr] != 0xFF;)
+				// read string
+				for (; ptr < data.Length && data[ptr] != 0xFF; ++ptr)
```

Comment:
the issue was the missing `ptr++` which is what openloco does. in the old code, I was doing this addition inside my while loop, and then subtracting 1 afterwards to correct for the `\0` character. i just changes the code to work like openloco's string table parsing and who would have thought, it just works